### PR TITLE
Improve documentation coverage for Herbert

### LINKED ISSUE #9035: Improve coverage of the documentation
Currently, some public classes are not documented anywhere because we didn't create the corresponding doc pages. Those missing pages are:

- Benchmark classes
- Bert Japanese
- Data collators

If someone feels like working on one of those, please tag yourself with a comment on this issue. Once the objects are properly documented, they can be removed from the `SHOULD_BE_DOCUMENTED` constant in [this file](https://github.com/huggingface/transformers/blob/1310e1a758edc8e89ec363db76863c771fbeb1de/utils/check_repo.py#L374).

### PR DESCRIPTION
**Pull request by Qbiwan**
# What does this PR do?

Fixes #9035

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

 @sgugger

### CODE REVIEW

File: docs/source/model_doc/herbert.rst [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+extensive evaluation, including several standard baselines and recently proposed, multilingual Transformer-based
+models.*
+
+Examples of use:
+
+.. code-block::
+
+  from transformers import XLMTokenizer, RobertaModel
+
+  tokenizer = XLMTokenizer.from_pretrained("allegro/herbert-klej-cased-tokenizer-v1")
```
Comment (LysandreJik) [inline docs/source/model_doc/herbert.rst:45]: Should these be `HerbertTokenizer`?

Comment (Qbiwan) [reply docs/source/model_doc/herbert.rst]: updated from XLMTokenizer to HerbertTokenizer

**Review by LysandreJik — APPROVED** [review]
LGTM! Thanks for working on this @Qbiwan!

**Comment by LysandreJik** [comment]
Still not letting me assign you @sgugger :(

File: docs/source/model_doc/herbert.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+extensive evaluation, including several standard baselines and recently proposed, multilingual Transformer-based
+models.*
+
+Examples of use:
+
+.. code-block::
+
+  from transformers import HerbertTokenizer, RobertaModel
+
+  tokenizer = HerbertTokenizer.from_pretrained("allegro/herbert-klej-cased-tokenizer-v1")
```
Comment (Qbiwan) [inline docs/source/model_doc/herbert.rst:45]: @LysandreJik  updated from XLMTokenizer to HerbertTokenizer

**Review by sgugger — APPROVED** [review]
Thanks for adding this!

**Pull request merged**